roles/ceph-config/tasks/main.yml

```diff
@@ -15,24 +15,38 @@
       num_osds: "{{ lvm_volumes | length | int }}"
     when: lvm_volumes | default([]) | length > 0
 
-  - name: run 'ceph-volume lvm batch --report' to see how many osds are to be created
-    ceph_volume:
-      cluster: "{{ cluster }}"
-      objectstore: "{{ osd_objectstore }}"
-      batch_devices: "{{ devices }}"
-      osds_per_device: "{{ osds_per_device | default(1) | int }}"
-      journal_size: "{{ journal_size }}"
-      block_db_size: "{{ block_db_size }}"
-      report: true
-      action: "batch"
-    register: lvm_batch_report
-    environment:
-      CEPH_VOLUME_DEBUG: 1
-      CEPH_CONTAINER_IMAGE: "{{ ceph_docker_registry + '/' + ceph_docker_image + ':' + ceph_docker_image_tag if containerized_deployment else None }}"
-      CEPH_CONTAINER_BINARY: "{{ container_binary }}"
-      PYTHONIOENCODING: utf-8
-    changed_when: false
-    when: devices | default([]) | length > 0
+  - block:
+      - name: look up for ceph-volume rejected devices
+        command: "{{ 'docker run --rm --ulimit nofile=1024:4096 --privileged=true -v /run/lvm/lvmetad.socket:/run/lvm/lvmetad.socket -v /var/run/udev/:/var/run/udev/:z -v /etc/ceph:/etc/ceph:z -v /dev:/dev --entrypoint=ceph-volume ' + ceph_docker_registry + '/' + ceph_docker_image + ':' + ceph_docker_image_tag if containerized_deployment else 'ceph-volume' }} inventory --format json"
+        register: rejected_devices
+
+      - name: set_fact rejected_devices
+        set_fact:
+          _rejected_devices: "{{ _rejected_devices | default([]) + [item.path] }}"
+        with_items: "{{ rejected_devices.stdout | from_json }}"
```

Comment:
looks like we need a default {} before the from_json due to the previous task that could be skipped (issue with python2)